stock_force_assign_by_type/views/stock_picking_views.xml

```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="view_picking_form_force_assign" model="ir.ui.view">
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_form" />
+        <field name="arch" type="xml">
+            <field name="use_create_lots" position="after">
+                <field name="picking_type_force_reservation" />
+            </field>
+            <!-- WARNING: The 'attrs' attributes of the 'do_unreserve' button and the
+                'forecast_availability' field are overridden in this view. This change
+                may impact visibility and functionality in other modules. Ensure thorough
+                review of migrations and dependencies. -->
+            <button name="do_unreserve" position="attributes">
```

Comment:
@pilarvargas-tecnativa Why this action should be obfuscated depending on force_reservation parameter ? IMHO the _should_bypass_reservation() override is sufficent.

Reply:
Depending on the type of operation in which the stock reservation is being forced, the button that performs the unreserve action is not available. For example, in a purchase order in which the picking type establishes a forced stock reservation, when we access the delivery note this action is not available (domain: ('parent.picking_type_code', '!=', 'outgoing')) nor do we have visibility of the reserved units (forecast_availability). I think it is necessary to have this option available in any operation where a forced stock reservation is made.

Reply:
Hi @rousseldenis the background context for this module is that we need to pay for something that goes out of our warehouse (the provider charges waste management fees), which is something unsupported by Odoo. So we need:

- To do a purchase order from which we can generate a vendor invoice.
- Only an incoming picking will complete the delivered (received) quantities in the PO.
- We need to reserve the right lots from our stock automatically as in a delivery picking (this is what this module allows us to do).
- We need to be able to re-assign/ unreserve as in a regular delivery. This is what this domain overrides allows us to do.